internal/pkg/cli/cli.go

```diff
@@ -30,8 +30,9 @@ func NewGlobalOpts() *GlobalOpts {
 	bindProjectName()
 
 	return &GlobalOpts{
-		projectName: viper.GetString(projectFlag),
-		prompt:      prompt.New(),
+		// Leave the projectName as empty in case it's overwritten by a global flag.
```

Comment:
I feel like since now we don't initiate the `opts.projectName` with `opts.ProjectName()`, which means `opts.projectName` and `opts.ProjectName()` they don't necessarily equal to each other, we need to also substitute all usage of `opts.ProjectName()` within `Ask()`, `Validate()`, and `Execute()` to `opts.projectName`. Otherwise if `-p` is set, `opts.ProjectName()` will get expired.

Reply:
I might be misunderstanding the comment 😅, but `opts.projectName` is always equal to `opts.ProjectName()` once `opts.ProjectName()` is invoked (the value is only set within `opts.ProjectName()`).

This change just doesn't set it in the constructor, so it's really the opposite: we want commands to use only `opts.ProjectName()`! 
When an object is created with `opts := NewGlobalOpts()`, the project name should be accessed with `opts.ProjectName()` so that it can grab the latest value from viper. By default the value is the name of the project in the workspace, but if a flag is provided then the value in viper is overwritten.

Let me know if that doesn't make sense, I can update the comments accordingly.

Reply:
I like @sonofachamp's idea of just using `viper` as the repository for our flags, that would allow us to remove this struct.

But that's another refactoring story :P

Reply:
I see :+1:

Reply:
Yeah, using `viper` directly might be more straightforward